src/asmcnc/apps/maintenance_app/widget_maintenance_z_misc_save.py

```diff
@@ -103,12 +103,14 @@ def get_info(self): # Localize me!
 
     def save(self):
 
```

Comment:
as per the conversation we had a little while back with arch, may be worth refactoring it to be more like: 

`if not save_touchplate_offset():`
`    warning popup `
`    return`
    
`if not save_z_head_maintenance():`
`    warning popup`
`    return`

`success popup`

Reply:
I made the change but it actually seems more convoluted now, can you check if it seems right?

Reply:
ah I think I caused confusion with my pseudo-code - if there's just one opportunity for there to be a touchplate offset popup, either in the save_touchplate_offset function or in the save function, then it won't need the show_popup flag. 

So can either do: 

`if not save_touchplate_offset`
`warning popup with 1-2mm info`
`return False`

and 

`def save_touchplate_offset`
`if out of range:`
`return False`

OR

`if not save_touchplate_offset`
`return False`

and 

`def save_touchplate_offset`
`if out of range:`
`warning popup with 1-2mm info`
`return False`

does that clarify things?

Reply:
But the flag is still surely needed, or else how will it know whether to show the general error popup or the out of range popup?

Reply:
ah I see what you mean - probably neatest to move the general error into the except of the save_touchplate_offset function I think

Reply:
But what about the case where the save fails without an exception being thrown?

Reply:
you're right - I missed that one. 

Ok, this works well then, the only other alternative would be to [make it throw an exception](https://stackoverflow.com/questions/2052390/manually-raising-throwing-an-exception-in-python), but I'm not sure that's any tidier in this circumstance.

Reply:
ok in that case your first solution was deffo tidier, sorry! 😅